File: library/src/main/java/me/proxer/library/api/LimitEndpoint.java
package me.proxer.library.api;

import javax.annotation.Nullable;

/**
 * Interface for all endpoints which allow limiting.
 *
 * @param <T> The type this endpoint returns.
 * @author Ruben Gees
 */
public interface LimitEndpoint<T> extends Endpoint<T> {

    /**
     * Sets a limit on the amount of items to return. Must be one or higher.
     */
    @SuppressWarnings("EmptyMethod")
    LimitEndpoint<T> limit(@Nullable Integer limit);
}

File: library/src/main/kotlin/me/proxer/library/api/LimitEndpoint.kt
package me.proxer.library.api

/**
 * Interface for all endpoints which allow limiting.
 *
 * @param <T> The type this endpoint returns.
 *
 * @author Ruben Gees
 */
interface LimitEndpoint<T> : Endpoint<T> {

    /**
     * Sets a limit on the amount of items to return. Must be one or higher.
     */
    fun limit(limit: Int?): LimitEndpoint<T>
}
